python/spear/env.py

```diff
@@ -448,26 +146,11 @@ def _get_step_info(self):
     def _reset(self):
         # reset the task first in case it needs to set the pose of actors,
         # then reset agent so it can refine the pose of actors
-        self._rpc_client.call("reset_task")
-        self._rpc_client.call("reset_agent")
+        self._instance.legacy_service.reset_task()
+        self._instance.legacy_service.reset_agent()
 
     def _is_ready(self):
-        return self._rpc_client.call("is_task_ready") and self._rpc_client.call("is_agent_ready")
-
-    def _get_random_points(self, num_points):
-        random_points = self._rpc_client.call("get_random_points", num_points)
-        return np.asarray(random_points, dtype=np.float64).reshape(num_points, 3)
-
-    def _get_random_reachable_points_in_radius(self, reference_points, search_radius):
-        assert reference_points.shape[1] == 3
-        reachable_points = self._rpc_client.call("get_random_reachable_points_in_radius", reference_points.flatten().tolist(), search_radius)
-        return np.asarray(reachable_points, dtype=np.float64).reshape(reference_points.shape)
-
-    def _get_paths(self, initial_points, goal_points):
-        assert initial_points.shape[1] == 3
-        assert goal_points.shape[1] == 3
-        paths = self._rpc_client.call("get_paths", initial_points.flatten().tolist(), goal_points.flatten().tolist())
-        return [ np.asarray(path, dtype=np.float64).reshape(-1, 3) for path in paths ]
+        return self._instance.legacy_service.is_task_ready() and self._instance.legacy_service.is_agent_ready()
 
 
 # metadata for describing a space including the shared memory objects
```

Comment:
Comment below still refers to `CoreUtils`

Reply:
Updated this comment.